backend/services/mailer.js

```diff
@@ -0,0 +1,46 @@
+const sgMail = require("@sendgrid/mail");
+const Joi = require("joi");
+
+function setKey(){
+  sgMail.setApiKey(process.env.SENDGRID_API_KEY);
+}
+setKey();
+const schema = Joi.object({
+    intention:Joi.string().valid('Invitation', 'F2A', 'Access Code').required(),
+    to: Joi.string().email().required(),
+    from: Joi.object({
+        email: Joi.string().email().required(),
+        name: Joi.string().required(),
+    }),
+    subject: Joi.string().required(),
+    text: Joi.string().required(),
+    html: Joi.string().required(),
```

Comment:
Just noticed when testing: when `html` is used, the content of `text` seems not used in email.